src/ripple/conditions/impl/Der.h

```diff
@@ -0,0 +1,2798 @@
+            return;
+        }
+        DerCoderTraits<T>::decode(decoder, v.col_);
+    }
+
+    static
+    std::uint64_t
+    length(
+        OctetStringCheckEqualSize<T> const& v,
+        boost::optional<GroupType> const& parentGroupType,
+        TagMode encoderTagMode,
+        TraitsCache& traitsCache)
+    {
+        return DerCoderTraits<T>::length(v.col_, parentGroupType, encoderTagMode, traitsCache);
+    }
+
+    static
+    int
+    compare(T const& lhs, T const& rhs, TraitsCache& traitsCache)
+    {
+        return DerCoderTraits<T>::compare(lhs, rhs, traitsCache);
+    }
+};
+
+/** DerCoderTraits for types that will be coded as "less size" constrained asn.1
+    octet strings
+
+    @see {@link #DerCoderTraits}
+*/
+template <class T>
+struct DerCoderTraits<OctetStringCheckLessSize<T>> : OctetStringTraits
+{
+    static
+    void
+    encode(Encoder& encoder, OctetStringCheckLessSize<T> const& v)
+    {
+        DerCoderTraits<T>::encode(encoder, v.col_);
+    }
+
+    static
+    void
+    decode(Decoder& decoder, OctetStringCheckLessSize<T>& v)
+    {
+        auto& slice = decoder.parentSlice();
+        if (slice.size() > v.constraint_)
+        {
+            // Return unsupported rather than contentLengthMismatch
+            // because this constraint is an implementation limit rather
+            // than a parser constraint
+            decoder.ec_ = make_error_code(Error::unsupported);
+            return;
+        }
+        DerCoderTraits<T>::decode(decoder, v.col_);
+    }
+
+    static
+    std::uint64_t
+    length(
+        OctetStringCheckLessSize<T> const& v,
+        boost::optional<GroupType> const& parentGroupType,
+        TagMode encoderTagMode,
+        TraitsCache& traitsCache)
+    {
+        return DerCoderTraits<T>::length(
+            v.col_, parentGroupType, encoderTagMode, traitsCache);
+    }
+
+    static
+    int
+    compare(T const& lhs, T const& rhs, TraitsCache& traitsCache)
+    {
+        return DerCoderTraits<T>::compare(lhs, rhs, traitsCache);
+    }
+};
+
+/** DerCoderTraits for std::bitset
+
+    bitsets will be coded as ans.1 bitStrings
+
+    @see {@link #DerCoderTraits}
+    @see {@link #make_octet_string_check_less}
+*/
+template <std::size_t S>
+struct DerCoderTraits<std::bitset<S>>
+{
+    constexpr static std::uint8_t mod8 = S % 8;
+    constexpr static std::uint8_t const minUnusedBits = mod8 ? 8 - mod8 : 0;
+    constexpr static std::uint8_t const maxBytes = mod8 ? 1 + S / 8 : S / 8;
+
+    constexpr static
+    GroupType
+    groupType()
+    {
+        return GroupType::bitString;
+    }
+
+    constexpr static
+    ClassId
+    classId()
+    {
+        return ClassId::universal;
+    }
+    
+    static
+    boost::optional<std::uint8_t> const&
+    tagNum()
+    {
+        static boost::optional<std::uint8_t> tn{3};
+        return tn;
+    }
+
+    static
+    std::uint8_t
+    tagNum(std::bitset<S> const&)
+    {
+        return 3;
```

Comment:
Fixed in `a028ad7 [fold] Responding to Ed PR feedback 1`

Also, an explanation: This is a traits class, so a function like `tagNum` that returns a "magic number" 3 is about the same as defining an enum that equals "3". I.e. it's not _that_ much of a magic number. However, putting them as enums does make it easier to audit all the constants at once, so I agree it's a good change.